interface/ceed-preconditioning.c

```diff
@@ -569,13 +573,20 @@ static int CeedSingleOperatorAssemble(CeedOperator op, CeedInt offset, CeedVecto
   // Get assembly data
   CeedOperatorAssemblyData data;
   CeedCall(CeedOperatorGetOperatorAssemblyData(op, &data));
-  const CeedEvalMode *eval_mode_in, *eval_mode_out;
-  CeedInt             num_eval_mode_in, num_eval_mode_out;
-  CeedCall(CeedOperatorAssemblyDataGetEvalModes(data, &num_eval_mode_in, &eval_mode_in, &num_eval_mode_out, &eval_mode_out));
-  CeedBasis basis_in, basis_out;
-  CeedCall(CeedOperatorAssemblyDataGetBases(data, &basis_in, NULL, &basis_out, NULL));
-
-  if (num_eval_mode_in == 0 || num_eval_mode_out == 0) {
+  const CeedEvalMode **eval_modes_in, **eval_modes_out;
+  CeedInt             *num_eval_modes_in, *num_eval_modes_out, num_active_bases;
+  CeedCall(CeedOperatorAssemblyDataGetEvalModes(data, &num_active_bases, &num_eval_modes_in, &eval_modes_in, NULL, &num_eval_modes_out,
+                                                &eval_modes_out, NULL, NULL));
+  CeedBasis *bases;
+  CeedCall(CeedOperatorAssemblyDataGetBases(data, NULL, &bases, NULL, NULL));
+  CeedBasis basis_in = bases[0];
+
+  if (num_active_bases > 1) {
+    // LCOV_EXCL_START
+    return CeedError(ceed, CEED_ERROR_UNSUPPORTED, "Cannot assemble operator with multiple active bases");
+    // LCOV_EXCL_STOP
+  }
```

Comment:
This means we still don't support full assembly for multiple bases, right?

Reply:
Correct. That will be much, much more work